UCT/API: Update uct_ep_am_zcopy description

dmitrygladkov opened the pull request:
## What

Update uct_ep_am_zcopy documentation

## Why ?

the `uct_ep_am_zcopy`description is not complete

## How ?

Added in the description for `uct_ep_am_zcopy`:
```
@return UCS_OK              Operation completed successfully.
```
```
@return UCS_ERR_NO_RESOURCE Could not start the operation now due to lack
                            of send resources.
```
```
@note If the operation returns @a UCS_INPROGRESS, the memory buffers pointed
      to by @a iov array must not be modified until the operation is completed
      by @a comp. @a header can be released or changed
```

yosefe commented on src/uct/api/uct.h:
```diff
@@ -2400,26 +2400,31 @@ UCT_INLINE_API ssize_t uct_ep_am_bcopy(uct_ep_h ep, uint8_t id,
+ * @return UCS_ERR_NO_RESOURCE Could not start the operation now due to lack
+ *                             of send resources.
+ *
+ * @note If the operation returns @a UCS_INPROGRESS, @a iov array must be valid and not
```
`If the operation returns @a UCS_INPROGRESS, the memory buffers pointed to by @a iov array must not be modified until the operation is completed by @a comp. @a header can be released or changed`

dmitrygladkov replied:
done

swx-jenkins commented:
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/8016/ for details.

mellanox-github commented:
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11114/ for details (Mellanox internal link).

mellanox-github commented:
<details><summary>Mellanox CI: <b>PASSED</b#  > on 33 workers (click for details)</summary>

**Note:** the logs will be deleted after 16-Jul-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/b8f7f18684eee5c81568faa2622241beffc24ea1/_main) | :heavy_check_mark: SUCCESS
[boo31_W0](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/df0afdcbfd54bf1e0382c807aa30c417cc16a67d/boo31_W0) | :heavy_check_mark: SUCCESS
[boo31_W1](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/2261ec23b2f32c2b2cceea2d47860ceb974d1e88/boo31_W1) | :heavy_check_mark: SUCCESS
[boo31_W2](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/74de72766260574709bcf962b6be70c18b24693f/boo31_W2) | :heavy_check_mark: SUCCESS
[boo31_W3](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/881de0c07d707e30f0a6d67014c72336b8cdb12a/boo31_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/df7c2ff155beba162f62dcd6e36721528327c6ed/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/9645e4e0c914c5b8fb353146e0e6605e00d76a43/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/d65a7a05658e38484525504e05d64d53bdc04a18/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/dd0f117a44d0213b47b681b08f70cd65494c8c4d/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/83773d80c720a4583239145b1907de06423824fc/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/c4f1d00169fb17fdf417ed4214e9104c60bdd0b9/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/c506c978100e38fd8e2e7bf28b2b4cd97d7247ca/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/e8531e97014a5b5dab7d76b8326693f5cbe73e8a/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/69a845cc196b51c28ec28343ddec8321447effa8/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/bd13b4bf2e1445aa4393414a27ff9861242fb7c6/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/19d2ac0afea4c3f1c4cd51dd854202b0c35f43a6/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/28d6262469104788800f7fa9c6e2ce6479bf1610/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/f9444fe832c1240d4091c05ebbde2d88ada36dbb/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/b8f6cc85b72b41c92c59bd6f11e5f0beca9c1d58/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/8989b0d6d0450fea31e90f3ded69c5c6f588287b/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/3fad7347fd030e21bbe1457c5e1b2c311d23b3d0/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/98366579d5eb76a65bfd8ac54c972db1d0103457/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/819730117064c22d1af91013f51c38723deae567/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/9648f1796203fc127772c711d181f7ed87282985/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/4deb1f55dcb17488a1b98863c09cbfd568291dca/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W0](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/3b406a1178894d887f5168bbe198480dcb4de856/hpc-test-node-upstream_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W1](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/06e9ee1d62cb74c8ab3903cf15ea1365a09d86d3/hpc-test-node-upstream_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W2](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/d7439e3de9b0b1657e2dae0d2f23cbc32515722f/hpc-test-node-upstream_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W3](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/ee071cba89f5d0984888ed3e30bcf6af580a585c/hpc-test-node-upstream_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/986eb063510ee0889328abfdffb5321dc7be1293/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/edd4279f36b8000aeaff6a2b006cf8c2d675b145/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/b346acecf33d2c91d1a08b41bb8b30a6cc4352c1/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/e062afea4f5cb9eb708b6a72d5b0e39f/raw/929646bf37c08a18537ceb2453d5db3a9c4119c4/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

swx-jenkins commented:
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/8022/ for details.

shamisp commented:
@tonycurtis please take a look

mellanox-github commented:
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11121/ for details (Mellanox internal link).

mellanox-github commented:
<details><summary>Mellanox CI: <b>PASSED</b#  > on 33 workers (click for details)</summary>

**Note:** the logs will be deleted after 16-Jul-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/a4bd8818aebb85dc76c593d4c8bfe8bb46a51c7b/_main) | :heavy_check_mark: SUCCESS
[boo31_W0](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/54ef1bbaec3dcd3365dcaf799a276cab6d62d455/boo31_W0) | :heavy_check_mark: SUCCESS
[boo31_W1](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/a29a3141402341808c79cb0a41d6f0799e0c2d35/boo31_W1) | :heavy_check_mark: SUCCESS
[boo31_W2](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/50f651c45e58bac996a6dc8d6e13fc7afde489aa/boo31_W2) | :heavy_check_mark: SUCCESS
[boo31_W3](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/5fe959de6e947a76bef2d3414a89d402d8eee06a/boo31_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/ca5956ea132271be16f2b22da78f8702685196d6/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/850cf4a4726f03b2ad3fcc4aa72a34c33aed6bff/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/d3520d3bc383bd9982ad695f2f3313f70f5c24e6/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/26e762f7c5daf6eb361b2c2a9a70e6299ec292e3/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/5dae6f14a116d1cb0e2346b2a1cdf0ee2a53f36a/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/7f078e5abd0a26380d851ee4c5be48220ec576be/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/4f387ab02baf192c18b711bf70d40e5c679731bd/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/6036871d2bde34de9213cdab2a898b883bba94ba/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/aed08be6a7d2fed8e1f220a931800d07e327b740/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/04dae8aa49cc422aef0eca74adc362b56b6dce16/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/6eece3dd21546ca3d72b627fd27d132bf2f06a33/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/215acd6cf912431d451dbd73d505154cfc4b9934/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/ee72f8bec2ee7092dacc2c90382f4cfd5202ac8b/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/312ed5c8acc6f7a6643da33f731d5c240992ec14/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/41bc48e3e6534a5266fdf55fa5318b397038e159/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/9e379bfe0d0268224bf18ed3538b688970578e86/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/a87e8152cbf607b0db659509453f276fe1c5dc99/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/6d1367f0c0a12ef7c03253092551a706c628ee26/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/6229982de24b37d7ddab639129b60e42bfb745b2/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/c72c817934a376473ecbfde6d2f28447bb3ae4be/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W0](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/796bc676356f4b5fad321e9043b01ecce120e8c8/hpc-test-node-upstream_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W1](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/4ec11bb5cd372cfb3e0f89a68313a125bf331b94/hpc-test-node-upstream_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W2](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/7f4ea2eb594b10cb75104f7bf4d75241d3fa4a53/hpc-test-node-upstream_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W3](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/ee71c265c89baea576e371bbf584e081cb5bacee/hpc-test-node-upstream_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/feeac3a4e08c4e52cc4385fef2038c567c4dd95b/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/fcee3e8a510e358645750906acc0be5653037d75/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/32f012e116eebe03938447a6b1caa1f20ce19c3a/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/d12dbf411c291893536169763f2e7a18/raw/3227e7921bdf47dd19e9912852fe401ab0ba552d/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

tonycurtis commented:
> uct.h:2407-2411

“a valid address” throughout.

“A particular structure must be a valid address”

Does that mean “all structures must be…” ?

“A NULL terminated…”.

Is that a NULL terminated “pointer" or “array"?  Array might be clearer here?

Period at end of description.

> uct.h:2423: 

> 
> Could not start the operation now due to lack 
> of send resources.
> 

Not sure about the word “now” here.  Completely open question: Does this imply the resources must be available on entry and without them, zcopy() can’t even get things going a bit later, and return INPROGRESS?

Tony

dmitrygladkov commented:
@tonycurtis 
> uct.h:2407-2411
> “a valid address” throughout. “A particular structure must be a valid address” Does that mean “all structures must be…” ? “A NULL terminated…”. Is that a NULL terminated “pointer" or “array"? Array might be clearer here?

This part of the description was not changed by me (only aligned, i.e. added whitespaces).

I've added only:
```
@return UCS_OK              Operation completed successfully.
```
```
@return UCS_ERR_NO_RESOURCE Could not start the operation now due to lack
                            of send resources.
```
```
@note If the operation returns @a UCS_INPROGRESS, the memory buffers pointed
      to by @a iov array must not be modified until the operation is completed
      by @a comp. @a header can be released or changed
```

> Period at end of description.
> uct.h:2423: Could not start the operation now due to lack of send resources.
> Not sure about the word “now” here. Completely open question: Does this imply the resources must be available on entry and without them, zcopy() can’t even get things going a bit later, and return INPROGRESS?

Yes, we can return `UCS_INPROGRESS`. But in some cases (TCP EP is in connection establishment), it's better to return `UCS_ERR_NO_RESOURCE`, because real AM Zcopy operation is not performed.

yosefe commented:
@dmitrygladkov can you pls fix the documentation according to @tonycurtis comment, even though it was not added by your PR? i think we can use this opportunity to improve our documentation :)

dmitrygladkov commented:
> can you pls fix the documentation according to @tonycurtis comment, even though it was not added by your PR?

Sure, I'll review other places as well to fix according to the @tonycurtis's comment

dmitrygladkov commented:
@tonycurtis @yosefe fixed the comments
could you review pls?

tonycurtis commented:
> On Jul 12, 2019, at 8:10 AM, Dmitry Gladkov <notifications@github.com> wrote:
> 
> @tonycurtis <https://github.com/tonycurtis> @yosefe <https://github.com/yosefe> fixed the comments
> could you review pls?
> 

Seems ok to me

swx-jenkins commented:
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/8070/ for details.

mellanox-github commented:
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11208/ for details (Mellanox internal link).

mellanox-github commented:
<details><summary>Mellanox CI: <b>FAILED</b> on 2 of 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 19-Jul-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/decce547c66235b9a0bdc94c6304d6177cbf8563/_main) | :x: FAILURE
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/10752e93cd9b677a80c33391c38f6ff00f6f34f8/hpc-test-node-new_W0) | :x: FAILURE
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/94b7179120218a318285a3187cb62c393e05c006/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/4fae99c873dc8773734f7677bbe9bbb2786b0781/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/6860d0cdf32632edff48228fa695df5c6df42b95/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/59e2ffc65f72f197fdbfc4bb2620cfbbdbcb98eb/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/ef0ad8b26bbbd96718048f1acbd8227e224069b0/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/5c32ac1d074e4b0926ee963aecde5473c41f44bb/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/d4cf009519336285c87b55cb8c8283496503c833/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/ee113a477e33f23899a9a712a014c94e13711aa8/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/a85da1b5ae65dfac1d4ce509dea0000d2ca8bdbc/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/791a7242a3e2ce34b55632d275666694ba01eeb9/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/f1b21c940e1bfc21c4c8f389a812bda309084008/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/9c63972b8701a441c1aaa296b20bb64d31d6e3aa/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/1f137690d59d17ea7328126e0372b7d01b9ed2f7/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/b2319c212d69ebef55b87b500f3bc0db31afa6f5/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/feb20ebe845ef2ff6f280eabd70b165080fd3cf0/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/735541c7bfc41765d3d9fb4af6901ecf1578cb10/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/d5300a94879ebcc1ca22e810733724b0d5c6b6ca/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/241ec9dc7dfe1ee2c1b47e342e2867808f20ed3b/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/f185ccc43d4df599a192d46c9fc3dc5539e463d3/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W0](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/ff092ae3bee7569ae3ec2938fe4e7d23c75f532d/hpc-test-node-upstream_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W1](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/4cf9efb0f008c241ae74495d29a748fbd650c4f4/hpc-test-node-upstream_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W2](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/ed34f395f7999fdfc024f40bfaf304cc6710f746/hpc-test-node-upstream_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W3](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/ed8335e082e00978c308a87fffeaebe49a0efa59/hpc-test-node-upstream_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/bd040daa0360cede04c77f8a727e7a13ee7923ef/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/d1e0758b409ba8e916985347e19e4e1c41d15771/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/beefaf6d14459e5f822c859ee6fb94234303bec9/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/9ee63b880ef135e15693a1c9eaa56439/raw/2c7d4edb23b84ee3aad142908240314a53b4ec19/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

dmitrygladkov commented:
bot:mlx:retest

mellanox-github commented:
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11218/ for details (Mellanox internal link).

mellanox-github commented:
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 19-Jul-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/25f660a72aab1906ec45818fd7045c1c630405da/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/dac065d67344d26c019ff98cd9fe27bf30684b84/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/87f04b4c2e2a431723f037f194a8d5919987706b/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/6d5e858200dd9d76513b5d0f986c98618975a081/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/175c5503d32255a37bc22c54d48b551d6ca4d738/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/5681fd42685c0f13de64792af7264783da196082/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/b60cb50b237422761202d9b167fc57d161904a7a/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/df35a64b5d671d3dd6128302fc29ee3c42325fac/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/b35ed93587b8b0ac5add5f8fb48869463abb2c02/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/e0962d9d383c58ec0dc1bca865e96d5026ae667c/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/a900ffe5d6a5f4a4eca285cfabb1a822c9c9f958/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/1047b83504d2db3d7002f2e1b017f10823062fb2/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/e441b6ef4ae3259761e04e020d0e254f333068b0/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/e9a3e7009454c37416f45c11c6dd049819ff0d22/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/4c306e5d14b3f4b78ef80c95ba6a5bfa31a33949/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/bbc4be7cef5862ebd4a68d043750e69c61957aed/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/0bc323be018aa0e8d0ac7566f474354bc4816010/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/efcd2c6fa0e0adc497c5ebe080b4fd7350bb3613/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/12589b792e9b07aa89719161e951f3796ecfd03a/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/d7208b5988679a5bfc4ab890fa4ccfb6d3af9106/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/cc8e7230d6745d33372b22b54920927582b6c615/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W0](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/16eebb2ac8bf73890af90b6f1a653dbb1bf49640/hpc-test-node-upstream_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W1](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/63ecf1cfa4bd95f67a4efce038343321dcef3803/hpc-test-node-upstream_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W2](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/05d986fd5daacbea7c627d031d8c97a7b185c45a/hpc-test-node-upstream_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W3](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/0bc25bc93c811ccd1ad1e381569e7b706edb165b/hpc-test-node-upstream_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/3bed42b0417d148bbaf3e1047418368a11114a8e/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/45cdeec43050da2f8a9463223c97621972305b82/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/1d9a8febccd830381d5830ffa22ffa545ea3571e/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/52041add315754165e7329eff2886f13/raw/f283ce077c0f095d36eaa1ba17ca2c977cfde440/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

swx-jenkins commented:
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/8081/ for details.

swx-jenkins commented:
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/8096/ for details.

dmitrygladkov commented:
@yosefe could you review pls?
